lib/logflare/backends/adaptor/postgres_adaptor/shared_repo.ex

```diff
@@ -0,0 +1,97 @@
+defmodule Logflare.Backends.Adaptor.PostgresAdaptor.SharedRepo do
+  use Ecto.Repo,
+    otp_app: :logflare,
+    adapter: Ecto.Adapters.Postgres
+
+  require Logger
+
+  alias Logflare.Source
+  alias Logflare.Backends.Backend
+
+  alias Logflare.Backends.Adaptor.PostgresAdaptor.Supervisor
+  alias Logflare.Backends.Adaptor.PostgresAdaptor.Repo.Migrations
+
+  def start(%Backend{} = backend) do
+    config = backend.config
+    pool_size = Application.get_env(:logflare, :postgres_backend_adapter)[:pool_size] || 10
+    schema = config["schema"] || config[:schema]
+
+    opts = [
+      url: config["url"] || config.url,
+      name: Supervisor.via(backend),
+      pool_size: pool_size,
+      # Wait until repo is fully up and running
+      sync_connect: true,
+      after_connect: {__MODULE__, :__after_connect__, [schema]}
+    ]
+
+    case Supervisor.start_child({__MODULE__, opts}) do
+      {:ok, pid} -> {:ok, pid}
+      # If there is already started process for this backend, then simply reuse
+      # it
+      {:error, {:already_started, pid}} -> {:ok, pid}
+      other -> other
```

Comment:
there shouldn't be any other case

Reply:
There should not, but just in case, I handle it there so it can be treated upstream if the requirements change in the future.